src/controller/CHIPDeviceController.cpp

```diff
@@ -1499,6 +1503,13 @@ void DeviceCommissioner::OnDeviceConnectedFn(void * context, OperationalDevicePr
                        ChipLogProgress(Controller, "Device connected callback with null pairing delegate. Ignoring"));
         commissioner->mPairingDelegate->OnPairingComplete(CHIP_NO_ERROR);
     }
+
+    // Release any CommissioneeDeviceProxies we have here as we now have an OperationalDeviceProxy.
```

Comment:
Shouldn't we do this even if `commissioner->mPairingDelegate == nullptr` and `commissioner->mCommissioningStage != CommissioningStage::kFindOperational`?  Right now we are taking an early return in that case...

Most simply, should this block be at the _top_ of the function?

Reply:
Ah, yes - missed the early return thanks to the VerifyOrReturn logic. Will remove and put it in a more sensible if /else function.

CommissioningStageComplete uses info from mDeviceBeingCommissioned, so the CommissioneeDeviceProxy cannot be removed until after that is complete.